spec/beaker/dsl/helpers/puppet_helpers_spec.rb

```diff
@@ -894,5 +894,92 @@ def stub_post_setup
+      subject.bounce_service( host, 'not_real_service', nil, 8000)
+    end
+  end
+
+  describe '#sleep_until_puppetdb_started' do
+    let( :options ) { Beaker::Options::Presets.new.presets }
+    before :each do
+      allow( subject ).to receive( :options ) { options }
+    end
+
+    it 'uses the default ports if none given' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8080/, anything(), anything() ).once.ordered
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8081/, anything() ).once.ordered
+      subject.sleep_until_puppetdb_started( host )
+    end
+
+    it 'allows setting the nonssl_port' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8084/, anything(), anything() ).once.ordered
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8081/, anything() ).once.ordered
+      subject.sleep_until_puppetdb_started( host, 8084 )
+
+    end
+
+    it 'allows setting the ssl_port' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8080/, anything(), anything() ).once.ordered
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8085/, anything() ).once.ordered
+      subject.sleep_until_puppetdb_started( host, nil, 8085 )
+    end
+  end
+
+  describe '#sleep_until_puppetserver_started' do
+    let( :options ) { Beaker::Options::Presets.new.presets }
+    before :each do
+      allow( subject ).to receive( :options ) { options }
+    end
+
+    it 'uses the default port if none given' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8140/, anything() ).once.ordered
+      subject.sleep_until_puppetserver_started( host )
+    end
+
+    it 'allows setting the port' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /8147/, anything() ).once.ordered
+      subject.sleep_until_puppetserver_started( host, 8147 )
+    end
+  end
+
+  describe '#sleep_until_nc_started' do
+    let( :options ) { Beaker::Options::Presets.new.presets }
+    before :each do
+      allow( subject ).to receive( :options ) { options }
+    end
+
+    it 'uses the default port if none given' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /4433/, anything() ).once.ordered
+      subject.sleep_until_nc_started( host )
+    end
+
+    it 'allows setting the port' do
+      host = hosts[0]
+      expect( subject ).to receive( :curl_with_retries ).with( anything(), anything(), /4435/, anything() ).once.ordered
+      subject.sleep_until_nc_started( host, 4435 )
+    end
+  end
```

Comment:
I'm generally pretty :-1: on mockist tests, these included. I agree we should be looking for acceptance tests here, though, to be fair, I'm roadblocking lots of things by not having some version of https://github.com/puppetlabs/beaker/pull/930 landed to make that easier for everyone.